documentation/course/e2/README.md

```diff
@@ -2159,7 +2164,53 @@ TODO @piterek130: Dodać diagram pakietów, opis architektury i endpointy.
 
 ## Logistyka
 
-TODO @mlodybercik: Dodać diagram pakietów, opis architektury i endpointy.
+Do implementacji systemu obsługującej logistykę JakPrzyjade wybrano język programowania **TypeScript** z użyciem frameworka **NestJS**, w środowisku **Node.js**.
+
+Wybór ten podyktowany był głównie brakiem skomplikowanych operacji wykonywanych przez ten wycinek systemu oraz powszechnością języka TypeScript (również w zespole). Rozszerzenie języka JavaScript o statyczne typowanie pozwala na uniknięcie wielu błędów poprzez przerzucenie ich dużej części na etap kodowania, co jest istotne w przypadku systemu, który ma być zarówno wydajny jak i bezpieczny.
+
+Framework **NestJS** został wybrany ze względu na swoje wsparcie statycznego typowania dzięki TypeScriptowi, modularność, wydajność, łatwość testowania oraz dużą ilość gotowych rozwiązań, które pozwolą na szybkie i efektywne tworzenie systemu dostarczającego funkcjonalność CRUD za pomocą REST API.
+
+![Diagram pakietów Leprechaun](./images/package-diagram-leprechaun.drawio.svg)
+
+Architektura podsystemu podzielona została na pionowe części zgodnie z architekturą _vertical slice_. Takie podejście stosuje się, gdy aplikacja jest podzielona na luźno powiązane ze sobą części, które są odpowiedzialne za różne aspekty systemu. Pozwala to na uniknięcie dużej ilości abstrakcji między warstwami logicznymi, co ułatwia zrozumienie kodu i jego rozwój. Taki wycinek systemu składa się z elementów wymaganych do jego działania, takich jak repozytoria, encje i kontrolery. W tym przypadku, każda z części składa się z elementów wchodzących w interakcję z bazą danych (`database`), elementów odpowiedzialnych za obsługę zapytań HTTP (`api`), elementów odpowiedzialnych za logikę biznesową (`domain`) oraz dodatkowego pakietu odpowiedzialnego za testy systemu (`test`). Wszystkie z nich korzystają z dodatkowego modułu dostarczającego funkcjonalności wspólne i pomocnicze (`shared`), zgodnie z logiką DRY.
```

Comment:
```suggestion
Architektura podsystemu podzielona została na pionowe części zgodnie z architekturą _vertical slice_. Takie podejście stosuje się, gdy aplikacja jest podzielona na luźno powiązane ze sobą części, które są odpowiedzialne za różne aspekty systemu. Pozwala to na uniknięcie dużej liczby abstrakcji między warstwami logicznymi, co ułatwia zrozumienie kodu i jego rozwój. Taki wycinek systemu składa się z elementów wymaganych do jego działania, takich jak repozytoria, encje i kontrolery. W tym przypadku, każda z części składa się z elementów wchodzących w interakcję z bazą danych (`database`), elementów odpowiedzialnych za obsługę zapytań HTTP (`api`), elementów odpowiedzialnych za logikę biznesową (`domain`) oraz dodatkowego pakietu odpowiedzialnego za testy systemu (`test`). Wszystkie z nich korzystają z dodatkowego modułu dostarczającego funkcjonalności wspólne i pomocnicze (`shared`), zgodnie z logiką DRY.
```

You can have 1 interface or 2, but you can't have a 1.5 interfaces. Same with abstract classes.